cirq/ops/common_gates.py

```diff
@@ -575,6 +576,11 @@ def __repr__(self) -> str:
             return 'CNOT'
         return 'CNOT**{!r}'.format(self.half_turns)
 
+    def on(self, *args: raw_types.QubitId, **kwargs: raw_types.QubitId) -> gate_operation.GateOperation:
+        if 'control' in kwargs.keys() and 'target' in kwargs.keys():
```

Comment:
Make sure to fail when a mix of named and positional arguments are specified:

```python
if args and kwargs:
    raise ValueError('Mixed named/positional control/target.')
```

Should also fail when there are unknown kwargs.